integration_test.go

```diff
@@ -0,0 +1,512 @@
+	for username, statements := range e.TestUsers {
+		if err := e.Elasticsearch.RevokeUser(context.Background(), statements, username); err != nil {
+			t.Fatal(err)
+		}
+		password := e.TestCreds[username]
+		if e.tc.Authenticate(t, username, password) {
+			t.Errorf("want authenication failure, got successful authentication for user:%s with password:%s", username, password)
+		}
+	}
+}
+
+func (e *IntegrationTestEnv) TestElasticsearch_RotateRootCredentials(t *testing.T) {
+	originalConfig := map[string]interface{}{
+		"username":    e.Username,
+		"password":    e.Password,
+		"url":         e.URL,
+		"ca_cert":     e.CaCert,
+		"client_cert": e.ClientCert,
+		"client_key":  e.ClientKey,
+	}
+	if !e.tc.Authenticate(t, e.Username, e.Password) {
+		t.Errorf("want successful authenication, got failed authentication for user:%s with password:%s", e.Username, e.Password)
+	}
+	configToStore, err := e.Elasticsearch.RotateRootCredentials(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if e.tc.Authenticate(t, e.Username, e.Password) {
+		t.Errorf("want authenication failure, got successful authentication for user:%s with password:%s", e.Username, e.Password)
+	}
+
+	if len(originalConfig) != len(configToStore) {
+		t.Fatalf("expected %s, received %s", originalConfig, configToStore)
+	}
+	for k, v := range originalConfig {
+		if k == "password" {
+			if configToStore[k] == v {
+				t.Fatal("password should have changed")
+			}
+			continue
+		}
+		if configToStore[k] != v {
+			t.Fatalf("for %s, expected %s but received %s", k, v, configToStore[k])
+		}
+	}
+}
+
+func readCertFile(t *testing.T, filename string) []byte {
+	t.Helper()
+	b, err := ioutil.ReadFile(filepath.Join("testdata", "certs", filename))
+	if err != nil {
+		t.Fatalf("Failed to read %s file: %s", filename, err)
+	}
+	return b
+}
+
+func prepareTestContainer(t *testing.T) (cleanup func(), client *http.Client, retAddress string) {
+	t.Helper()
+
+	certsdir, err := filepath.Abs(filepath.Join("testdata", "certs"))
+	if err != nil {
+		t.Fatalf("could not create an absolute path to the testdata/certs directory: %s", err)
+	}
+
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		t.Fatalf("Failed to connect to docker: %s", err)
+	}
+
+	var env = []string{
+		"discovery.type=single-node",
+		"xpack.security.enabled=true",
+		"xpack.license.self_generated.type=trial",
+		"xpack.security.http.ssl.enabled=true",
+		"xpack.security.http.ssl.key=/usr/share/elasticsearch/config/certificates/server-key.pem",
+		"xpack.security.http.ssl.certificate=/usr/share/elasticsearch/config/certificates/server.pem",
+		"xpack.security.http.ssl.certificate_authorities=/usr/share/elasticsearch/config/certificates/rootCA.pem",
+		"xpack.security.http.ssl.client_authentication=required",
+		"ELASTIC_PASSWORD=" + esInitialPassword,
+	}
+
+	dockerOptions := &dockertest.RunOptions{
+		Repository: "docker.elastic.co/elasticsearch/elasticsearch",
+		Tag:        "7.3.0",
```

Comment:
It's a shame they don't support the "latest" tag because then we could have the tests start failing if a new version came out that this code wasn't compatible with, which would be super informative!